src/masternodes/rpc_tokens.cpp

```diff
@@ -815,6 +815,155 @@ UniValue minttokens(const JSONRPCRequest& request) {
     return signsend(rawTx, pwallet, optAuthTx)->GetHash().GetHex();
 }
 
+UniValue decodecustomtx(const JSONRPCRequest& request)
+{
+    RPCHelpMan{"decodecustomtx",
+        "\nGet detailed information about a DeFiChain custom transaction.\n"
+        "can be enabled to return details for any transaction.",
+        {
+            {"hexstring", RPCArg::Type::STR_HEX, RPCArg::Optional::NO, "The transaction hex string"},
+            {"iswitness", RPCArg::Type::BOOL, /* default */ "depends on heuristic tests", "Whether the transaction hex is a serialized witness transaction.\n"
+                "If iswitness is not present, heuristic tests will be used in decoding.\n"
+                "If true, only witness deserialization will be tried.\n"
+                "If false, only non-witness deserialization will be tried.\n"
+                "This boolean should reflect whether the transaction has inputs\n"
+                "(e.g. fully valid, or on-chain transactions), if known by the caller."
+            },
+        },
+        RPCResult{
+            "{\n"
+            "  \"type\":               (string) The transaction type.\n"
+            "  \"valid\"               (bool) Whether the transaction was valid.\n"
+            "  \"results\"             (json object) Set of results related to the transaction type\n"
+            "  \"block height\"        (string) The block height containing the transaction.\n"
+            "  \"blockhash\"           (string) The block hash containing the transaction.\n"
+            "  \"confirmations\": n,   (numeric) The number of confirmations for the transaction."
+            "  \"warnings\":           (string) any network and blockchain warnings." 
+            "}\n"
+        },
+        RPCExamples{
+            HelpExampleCli("decodecustomtx", "\"hexstring\"")
+            + HelpExampleRpc("decodecustomtx", "\"hexstring\"")
+        },
+    }.Check(request);
+
+    RPCTypeCheck(request.params, {UniValue::VSTR, UniValue::VBOOL});
+
+    CMutableTransaction mtx;
+
+    bool try_witness = request.params[1].isNull() ? true : request.params[1].get_bool();
+    bool try_no_witness = request.params[1].isNull() ? true : !request.params[1].get_bool();
+
+    if (!DecodeHexTx(mtx, request.params[0].get_str(), try_no_witness, try_witness)) {
+        throw JSONRPCError(RPC_DESERIALIZATION_ERROR, "TX decode failed");
+    }
+
+    int nHeight{0};
+    bool actualHeight{false};
+    CustomTxType guess;
+    UniValue txResults(UniValue::VOBJ);
+    Res res{};
+    uint256 hashBlock;
+    CBlockIndex* blockindex{nullptr};
+    CTransactionRef tx = MakeTransactionRef(std::move(mtx));
+    std::string warnings;
+
+    if (tx)
```

Comment:
Actually also considering suggesting removing this section, and making this function only return the parameters of the tx, not the result (As if we have tx hash we can just call getcustomtx for that anyway).